src/libraries/System.Text.Encoding/tests/Encoding/TranscodingStreamTests.cs

```diff
@@ -0,0 +1,893 @@
+            Assert.Throws<ObjectDisposedException>(() => transcodingStream.WriteByte((byte)'x'));
+        }
+
+        [Fact]
+        public void Dispose_WithLeaveOpenFalse_DisposesInnerStream()
+        {
+            // Sync
+
+            MemoryStream innerStream = new MemoryStream();
+            Stream transcodingStream = Encoding.CreateTranscodingStream(innerStream, Encoding.UTF8, Encoding.UTF8, leaveOpen: false);
+            transcodingStream.Dispose();
+            transcodingStream.Dispose(); // calling it a second time should no-op
+            Assert.Throws<ObjectDisposedException>(() => innerStream.Read(Span<byte>.Empty));
+
+            // Async
+
+            innerStream = new MemoryStream();
+            transcodingStream = Encoding.CreateTranscodingStream(innerStream, Encoding.UTF8, Encoding.UTF8, leaveOpen: false);
+            transcodingStream.DisposeAsync().GetAwaiter().GetResult();
+            transcodingStream.DisposeAsync().GetAwaiter().GetResult(); // calling it a second time should no-op
+            Assert.Throws<ObjectDisposedException>(() => innerStream.Read(Span<byte>.Empty));
+        }
+
+        [Fact]
+        public void Dispose_WithLeaveOpenTrue_DoesNotDisposeInnerStream()
+        {
+            // Sync
+
+            MemoryStream innerStream = new MemoryStream();
+            Stream transcodingStream = Encoding.CreateTranscodingStream(innerStream, Encoding.UTF8, Encoding.UTF8, leaveOpen: true);
+            transcodingStream.Dispose();
+            transcodingStream.Dispose(); // calling it a second time should no-op
+            innerStream.Read(Span<byte>.Empty); // shouldn't throw
+
+            // Async
+
+            innerStream = new MemoryStream();
+            transcodingStream = Encoding.CreateTranscodingStream(innerStream, Encoding.UTF8, Encoding.UTF8, leaveOpen: true);
+            transcodingStream.DisposeAsync().GetAwaiter().GetResult();
+            transcodingStream.DisposeAsync().GetAwaiter().GetResult(); // calling it a second time should no-op
+            innerStream.Read(Span<byte>.Empty); // shouldn't throw
+        }
+
+        [Fact]
+        public void Flush_FlushesInnerStreamButNotDecodedState()
+        {
+            // Arrange
+
+            CancellationToken expectedCancellationToken = new CancellationTokenSource().Token;
+            Task expectedFlushAsyncTask = Task.FromResult("just some task");
+
+            var innerStreamMock = new Mock<MemoryStream>() { CallBase = true };
+            innerStreamMock.Setup(o => o.FlushAsync(expectedCancellationToken)).Returns(expectedFlushAsyncTask);
+            Stream transcodingStream = Encoding.CreateTranscodingStream(innerStreamMock.Object, Encoding.UTF8, Encoding.UTF8, leaveOpen: true);
+
+            transcodingStream.Write(new byte[] { 0x7A, 0xE0 });
+            innerStreamMock.Verify(o => o.Flush(), Times.Never);
+            innerStreamMock.Verify(o => o.FlushAsync(It.IsAny<CancellationToken>()), Times.Never);
```

Comment:
I thought `Times.Never` (and others) were functions: https://github.com/moq/moq4/blob/master/src/Moq/Times.cs#L158?

Reply:
There are convenience overloads of `Mock<T>.Verify` (see https://github.com/moq/moq4/blob/8cca0e839a6131e666f9d0b8857bc5802a92f8f4/src/Moq/Mock.Generic.cs#L773) that allow this syntax to work.